test/lib/dialogs/addon_selector_test.rb

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) [2017-2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+#
+require_relative "../../test_helper"
+
+require "cwm/rspec"
+require "y2packager/product_location"
+require "y2packager/product_location_details"
+require "y2packager/dialogs/addon_selector"
+
+describe Y2Packager::Dialogs::AddonSelector do
+  subject { described_class.new(products) }
+
+  include_examples "CWM::Dialog"
+
+  let(:basesystem) do
+    Y2Packager::ProductLocation.new(
+      "SLE-15-Module-Basesystem 15.0-0",
+      "/Basesystem",
+      product: Y2Packager::ProductLocationDetails.new(product: "sle-module-basesystem")
```

Comment:
NP: It's a bit confusing to have a param named `product` which actually expects a `ProductLocation` in the `Product` constructor. Just a note for the future, if it could be helpful.